File: samples/client/3_1_0_unit_test/java/src/main/java/org/openapijsonschematools/client/components/schemas/NullTypeMatchesOnlyTheNullObject.java
package org.openapijsonschematools.client.components.schemas;
import org.checkerframework.checker.nullness.qual.Nullable;
import org.openapijsonschematools.client.schemas.NullJsonSchema;

public class NullTypeMatchesOnlyTheNullObject extends NullJsonSchema {
    // nest classes so all schemas and input/output classes can be public
    
    
    public static class NullTypeMatchesOnlyTheNullObject1 extends NullJsonSchema.NullJsonSchema1 {
        private static @Nullable NullTypeMatchesOnlyTheNullObject1 instance = null;
        public static NullTypeMatchesOnlyTheNullObject1 getInstance() {
            if (instance == null) {
                instance = new NullTypeMatchesOnlyTheNullObject1();
            }
            return instance;
        }
    }

}

File: samples/client/3_1_0_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/components/schemas/NullTypeMatchesOnlyTheNullObject.kt
package org.openapijsonschematools.client.components.schemas
import org.openapijsonschematools.client.schemas.NullJsonSchema

class NullTypeMatchesOnlyTheNullObject: NullJsonSchema() {
    // nest classes so all schemas and input/output classes can be public
    
    
    class NullTypeMatchesOnlyTheNullObject1 : NullJsonSchema.NullJsonSchema1() {
        companion object {
            @Volatile
            private var instance: NullTypeMatchesOnlyTheNullObject1? = null
    
            fun getInstance() =
                instance ?: synchronized(this) {
                    instance ?: NullTypeMatchesOnlyTheNullObject1().also { instance = it }
                }
        }
    }

}
